src/OrchardCore.Modules/OrchardCore.Layers/Services/LayerFilter.cs

```diff
@@ -146,5 +152,18 @@ public async Task OnResultExecutionAsync(ResultExecutingContext context, ResultE
 
             await next.Invoke();
         }
+
+        private async Task<IEnumerable<LayerMetadata>> FilterWidgetsByCultureAsync(IEnumerable<LayerMetadata> widgets)
+        {
+            var widgetsWithCulture = await Task.WhenAll(widgets.Select(async w =>
+            {
```

Comment:
Please don't use `Task.WhenAll()` when it triggers the execution of scoped services.